### ISSUE TITLE
Unable to reach USMT binaries - Windows 10 issue

### ISSUE THREAD

**Issue by EddieRio** [nickrod518/Create-EXEFromPS1#3]
No saved state found at [C:\TEMP\MigrationStore].
Unable to reach USMT binaries. Verify [C:\Users\xxx\AppData\Local\Temp\IXP000.TMP\USMT\amd64] exists and restart script.

It's not copying/creating the USMT folder in the temp folder

**Comment by EddieRio** [comment]
# Default configuration options - make edits here
# Default domain to use for profile creation
$DefaultDomain = 'xxx'

# Default accounts to exclude from migration in the form of "Domain\UserName"
$DefaultExcludeProfile = @(
    "$env:COMPUTERNAME\default*",
    "NT Service\*"
)

# By default local accounts that don't exist on the new computer will not be created for security measures
# To create these accounts set this to true
$DefaultLACreate = $false

# By default local accounts that are created from the previous option will be disabled for security measures
# To enable these accounts set this to true
$DefaultLACEnable = $false

# Default password for accounts created by previous two options
$DefaultLAPassword = 'P@ssw0rd!'

# Use this to disallow migrations on IP's other than what's specified
$ValidIPAddress = '*'

# Path to store the migration data on the new computer, directory will be created if it doesn't exist
$MigrationStorePath = 'C:\TEMP\MigrationStore'

# Default user profile items to exclude from migration, more info found here: 
# https://technet.microsoft.com/en-us/library/cc722303(v=ws.10).aspx
$DefaultIncludeAppData = $true
$DefaultIncludeLocalAppData = $false
$DefaultIncludePrinters = $true
$DefaultIncludeRecycleBin = $false
$DefaultIncludeMyDocuments = $true
$DefaultIncludeWallpapers = $true
$DefaultIncludeDesktop = $true
$DefaultIncludeFavorites = $true
$DefaultIncludeMyMusic = $true
$DefaultIncludeMyPictures = $true
$DefaultIncludeMyVideo = $true

# Default extra directories to include
$DefaultExtraDirectories = @()

# Default recent profiles
$DefaultRecentProfiles = $true
$DefaultRecentProfilesDays = 90

# Get USMT binary path according to OS architecture. If you used the zip provided, unzip in the same directory as this script
if ((Get-WmiObject Win32_OperatingSystem).OSArchitecture -match '64') { 
    $USMTPath = "$ScriptRoot\USMT\amd64"
} else { 
    $USMTPath = "$ScriptRoot\USMT\x86"
}

# Define whether to continue on errors such as file already exists during restore or read issue during capture.
$ContinueOnError = $True

<# Define the level of verbosity for the USMT scan/load commands
0 Only the default errors and warnings are enabled.
1 Enables verbose output.
4 Enables error and status output.
5 Enables verbose and status output.
8 Enables error output to a debugger.
9 Enables verbose output to a debugger.
12 Enables error and status output to a debugger.
13 Enables verbose, status, and debugger output. #>
$VerboseLevel = 13

# Define how to handle EFS format files. Options are abort (default behaviour), skip, decryptcopy, copyraw
$EFSHandling = "abort"

# Users to additionially send every migration result to
$DefaultEmailEnabled = $false
$DefaultEmailSender = 'MigrationAlert@company.com'
$DefaultEmailRecipients = @('my.email@company.com')
$DefaultSMTPServer = 'smtp.domain.local'

# LastLogin query when gathering profiles - disabling will speed up profile search
$QueryLastLogon = $false

**Comment by EddieRio** [comment]
[USMT.txt](https://github.com/nickrod518/Create-EXEFromPS1/files/2275658/USMT.txt)

**Comment by nickrod518** [comment]
Okay that appears to be the config file from one of my other projects. What is the command you're using to build that exe?

**Comment by EddieRio** [comment]
It works fine running the Invoke-USMTGUI.ps1, but after compiling it gets the error message and the migrate button is disabled.

I'm running New-EXEFromPS1 -SelectSupplementalFiles

![migrate button](https://user-images.githubusercontent.com/15014782/43921376-91cc90ba-9be9-11e8-876d-9ff2f1cda075.JPG)

**Comment by nickrod518** [comment]
Try the select directory option instead of the select files option.

**Comment by EddieRio** [comment]
It's kind working now, the migrate button is enable but the data to include are unchecked :-)
![migrate button and options](https://user-images.githubusercontent.com/15014782/43922363-86c7e2f2-9bec-11e8-989a-f7d896740fb0.JPG)

**Comment by EddieRio** [comment]
Works great running the ps1 manually

![running ps1 manually](https://user-images.githubusercontent.com/15014782/43922422-b687dd44-9bec-11e8-9b67-0612cb916498.JPG)

**Comment by nickrod518** [comment]
Can you doublecheck the config that's included in the version you're turning into an exe has your latest changes? Or can you additionally use the select files flag to select the config file?

**Comment by nickrod518** [comment]
Sorry for not being more clear. What I mean is that you should try re-running the New-EXEFromPs1 command using the SelectSupplementalDirectory to select the USMT folder and the SelectSupplementalFile to select the Config.txt file.

**Comment by nickrod518** [comment]
Sorry for not realizing this sooner... the script supports only ONE of the following parameters:
SpecifyFiles, SpecifyDirectory, SelectFiles, SelectDirectory

You won't be able to use more than one. I'd suggest using SpecifyDirectory and include all related scripts in that directory. Something like a "USMT" directory that contains both the config and the USMT binaries folder. You may need to modify the path to the USMT binaries within your config to account for this.

**Comment by NathanTheGr8** [comment]
I was just reading through your code and figured this out. My plan is to move the Config.ps1 file into the USMT folder. I will also edit the Invoke-USMTGUI.ps1 file to point to the new Config.ps1 

Line 25
```
    # Load the options in the Config file
	. "$ScriptRoot\USMT\Config.ps1"
```

**Comment by nickrod518** [comment]
Yep, that should do it.

**Comment by NathanTheGr8** [comment]
Cool, I just got it to work. For future people. To package Migrate-WindowsUserProfile you just need to

1. Move the Config.ps1 file into the USMT folder
2. Edit Invoke-USMTGUI.ps1 line 25 to import the new config file location
3. Run This command to package it
```
New-EXEFromPS1 -PSScriptPath "PathTo\Invoke-USMTGUI.ps1" -SelectSupplementalDirectory
```
4. Select the USMT folder
5. Done! The Exe will be in the same location as the Invoke-USMTGUI.ps1 file.

I did not have to edit the USMT path in the Config.ps1. I think this is because you are importing Config.ps1 into the Invoke-USMTGUI.ps1 file, so the $ScriptRoot variable evaluates to the location of Invoke-USMTGUI.ps1 instead of Config.ps1's location.

How the exe has to unzip the attached USMT folder adds a long delay to the application start time, so I may try to package it a different way. Still a really cool tool.
